lms/djangoapps/courseware/module_render.py

```diff
@@ -554,12 +554,18 @@ def handle_deprecated_progress_event(block, event):
             if requested_user_id != user.id:
                 log.warning("{} tried to submit a completion on behalf of {}".format(user, requested_user_id))
                 return
-            BlockCompletion.objects.submit_completion(
-                user=user,
-                course_key=course_id,
-                block_key=block.scope_ids.usage_id,
-                completion=1.0,
-            )
+
+            # If blocks explicitly declare support for the new completion API,
+            # we expect them to emit 'completion' events,
+            # and we ignore the deprecated 'progress' events
+            # in order to avoid duplicate work and support slightly different semantics.
```

Comment:
Nit: support doesn't quite fit grammatically.  I would say "in order to avoid duplicate work and possibly conflicting semantics."

Reply:
👍